ksql-execution/src/main/java/io/confluent/ksql/execution/util/ExpressionTypeManager.java

```diff
@@ -216,8 +217,23 @@ public Void visitDereferenceExpression(
         final DereferenceExpression node,
         final ExpressionTypeContext expressionTypeContext
     ) {
-      throw new IllegalArgumentException("Dereferenced expressions should have been rewritten to "
-          + FetchFieldFromStruct.FUNCTION_NAME + " by this point");
+      process(node.getBase(), expressionTypeContext);
```

Comment:
This is called to determine the result type of an `DerferenceExpression`.

Previously, it wouldn't be called because the `DerferenceExpression` would have been re-written as a `FunctionCall`, but notice that the `visitFunctionCall` had explicit special handling of `FetchFieldFromStruct` function calls.

IMHO, it's cleaner just handling the call here.